src/Testing/PrefixedResourceIdTestCase.php

```diff
@@ -36,25 +36,27 @@ public function invalidMollieResourceIds(): Generator
     public function itCannotBeConstructedFromAnInvalidMollieResourceIdString(string $value): void
     {
         $this->expectExceptionObject(
-            call_user_func([$this->expectedExceptionClass(), 'missingExpectedPrefix'], $value, $this->expectedPrefix())
+            call_user_func([$this::expectedExceptionClass(), 'missingExpectedPrefix'], $value, $this::expectedPrefix())
         );
 
-        $this->resourceIdClass()::fromString($value);
+        $this::resourceIdClass()::fromString($value);
     }
 
-    public function validMollieResourceIds(): Generator
+    public static function validMollieResourceIds(): Generator
     {
-        yield 'Short identifier' => [$this->expectedPrefix() . Str::random(4)];
-        yield 'Long identifier' => [$this->expectedPrefix() . Str::random(8)];
+        yield 'Short identifier' => [fn (self $test): string => $test::expectedPrefix() . Str::random(4)];
+        yield 'Long identifier' => [fn (self $test): string => $test::expectedPrefix() . Str::random(8)];
```

Comment:
Why wrap the expectedPrefix generation in a closure? Can't we do this instead?
```suggestion
        yield 'Short identifier' => [self::expectedPrefix() . Str::random(4)];
        yield 'Long identifier' => [self::expectedPrefix() . Str::random(8)];
```

Reply:
I reckon the tests run for setting up the amount of tests that will run etc. before actually running the tests. It tries to call an abstract method which is not possible.